toolkit.nu

```diff
@@ -1,47 +1,101 @@
-# pull down the latest nightly build of Nushell
+use std log
+
+# install the latest nightly build of Nushell
 #
 # this command will
 # - get the metadata of the latest build of Nushell in the nightly repo
-# - filter the assets that match the search pattern `target`
-# - fuzzy-ask one of them or use the single match
+# - interactively ask for one of them or use architecture-specific defaults
 # - download the archive
-# - give some hints about the version and the hash and how to extract the archive
+# - extract the archive
+# - install the `nu` binary
 export def get-latest-nightly-build [
-    target: string = "" # the target architecture, matches all of them by default
+    --install-dir: path = "~/.local/bin/" # the directory where to install the `nu` binary
+    --interactive # ask for the architecture to install interactively
+    --full # install Nushell with all extra features enabled
+    --musl # install Nushell from the MUSL builds (linux x86_64 only)
 ]: nothing -> nothing {
     let latest = http get https://api.github.com/repos/nushell/nightly/releases
         | sort-by published_at --reverse
         | first
 
-    let matches = $latest.assets
+    let assets = $latest.assets
         | get name
-        | where $it =~ $target
-        | where $it !~ 'msi'
+        | where not ($it | str ends-with ".msi")
         | parse --regex 'nu-\d\.\d+\.\d-(?<arch>[a-zA-Z0-9-_]*)\..*'
         | get arch
 
-    let arch = match ($matches | length) {
-        0 => {
-            let span = metadata $target | get span
-            error make {
-                msg: $"(ansi red_bold)no_match(ansi reset)"
-                label: {
-                    text: $"no architecture matching this in ($latest.html_url)"
-                    start: $span.start
-                    end: $span.end
+    let arch = if $interactive {
+        match ($assets | length) {
+            0 => {
+                error make --unspanned {
+                    msg: (
+                          $"(ansi red_bold)unexpected_internal_error(ansi reset):\n"
+                        + "no nightly build..."
+                    )
+                }
+            },
+            1 => { $assets.0 },
+            _ => {
+                let choice = $assets | input list --fuzzy $"Please (ansi cyan)choose one architecture(ansi reset):"
+                if ($choice | is-empty) {
+                    print "user chose to exit"
+                    return
                 }
+
+                $choice
+            },
+        }
+    } else {
+        if $musl and (($nu.os-info.arch != "x86_64") or ($nu.os-info.name != "linux")) {
```

Comment:
musl target can runs on aarch64 too, we just build failed now and may add it some day: https://github.com/nushell/nushell/issues/10444

Reply:
but for now there's no support right, so i'd say we should add that later once the architecture supports MUSL ? :yum: